system/classes/KO7/I18n.php

```diff
@@ -65,75 +74,100 @@ public static function lang($lang = NULL)
 	 *
 	 *     $hello = I18n::get('Hello friends, my name is :name');
 	 *
-	 * @param   string  $string text to translate
-	 * @param   string  $lang   target language
+	 * @param   string|array  $string Text to translate or array [text, values]
+	 * @param   string  	  $lang   Target Language
+	 * @param   string		  $source Source Language
 	 * @return  string
 	 */
-	public static function get($string, $lang = NULL)
+	public static function get($string, string $lang = NULL, string $source = NULL)
 	{
+		$values = [];
+
+		// Check if $string is array [text, values]
+		if (Arr::is_array($string))
+		{
+			if (isset($string[1]) && Arr::is_array($string[1]))
+			{
+				$values = $string[1];
+			}
+			$string = $string[0];
+		}
+
+		// Set Target Language if not set
 		if ( ! $lang)
 		{
 			// Use the global target language
 			$lang = I18n::$lang;
 		}
 
-		// Load the translation table for this language
-		$table = I18n::load($lang);
+		// Set source Language if not set
+		if ( ! $source)
+		{
+			// Use the global source language
+			$source = I18n::$source;
+		}
+
+		// Load Table only if Source language does not match target language
+		if ($source !== $lang)
+		{
+			// Load the translation table for this language
+			$table = I18n::load($lang);
 
-		// Return the translated string if it exists
-		return isset($table[$string]) ? $table[$string] : $string;
+			// Return the translated string if it exists
+			$string = $table[$string] ?? $string;
+		}
+
+		return empty($values) ? $string : strtr($string, $values);
 	}
 
 	/**
 	 * Returns the translation table for a given language.
 	 *
-	 *     // Get all defined Spanish messages
```

Comment:
Spanish? I think there is a mixure, 2 lines later en-us

Reply:
Thx :)